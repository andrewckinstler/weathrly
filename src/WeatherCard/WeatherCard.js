import React from 'react';
import moment from 'moment';
import './WeatherCard.scss'


export const WeatherCard = props => {
  const { date, weather, main } = props
  const icon = weather[0].icon
  const min = main.temp_min.toFixed(0);
  const max = main.temp_max.toFixed(0)

  return (
    <article className='card'>
      <h3>{moment(date).format('dddd l')}</h3>
      <img src={`http://openweathermap.org/img/wn/${icon}@2x.png`}  alt='condition icon'/>
      <p><span className='bold'>H: </span>{min}°/ <span className='bold'>L: </span>{max}°</p>
    </article>
  )
}