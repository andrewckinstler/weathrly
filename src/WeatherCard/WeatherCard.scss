.card {
  padding: 15px;
  border-radius: 10px;
  background-color: #fff;
  text-align: center;

  img { 
    padding: 7px 0;
  }
  
  .bold {
    font-weight: bold;
  }
}