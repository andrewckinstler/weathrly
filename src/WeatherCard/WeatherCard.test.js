import { shallow } from 'enzyme';
import React from 'react';
import { WeatherCard } from './WeatherCard';

describe('Weather card', () => {
  it('should match a snapshot', () => {

    const dt = '12/30';
    const weather = [{icon: '10d'}]
    const main = { temp_max: 3, temp_min: 0 }


    const wrapper = shallow(<WeatherCard key={1} date={dt} weather={weather} main={main}/>)
    
    expect(wrapper).toMatchSnapshot()
  })
})