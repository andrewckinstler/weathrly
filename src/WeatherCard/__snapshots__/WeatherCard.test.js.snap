// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather card should match a snapshot 1`] = `
<article
  className="card"
>
  <h3>
    Sunday 12/30/2001
  </h3>
  <img
    alt="condition icon"
    src="http://openweathermap.org/img/wn/10d@2x.png"
  />
  <p>
    <span
      className="bold"
    >
      H: 
    </span>
    0
    °/ 
    <span
      className="bold"
    >
      L: 
    </span>
    3
    °
  </p>
</article>
`;
